<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BootEntryForm</class>
 <widget class="QWidget" name="BootEntryForm">
  <property name="windowTitle">
   <string>Boot entry form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QLabel" name="error_text">
     <property name="toolTip">
      <string>Error</string>
     </property>
     <property name="statusTip">
      <string>Error</string>
     </property>
     <property name="text">
      <string notr="true"/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="error_note">
     <property name="toolTip">
      <string>Error note</string>
     </property>
     <property name="statusTip">
      <string>Error note</string>
     </property>
     <property name="text">
      <string>This entry placeholder is shown here to indicate it's referenced in boot order. It won't be modified on save, just left as is.</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="form_fields" native="true">
     <layout class="QFormLayout" name="form_fields_layout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::ExpandingFieldsGrow</enum>
      </property>
      <property name="labelAlignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="index_label">
        <property name="toolTip">
         <string>Index</string>
        </property>
        <property name="statusTip">
         <string>Index</string>
        </property>
        <property name="text">
         <string>Index</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="index_text">
        <property name="font">
         <font>
          <family>Monospace</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Index</string>
        </property>
        <property name="statusTip">
         <string>Index</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Entry index.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="inputMask">
         <string notr="true">&gt;\0\xHHHH;_</string>
        </property>
        <property name="text">
         <string notr="true">0x</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="description_label">
        <property name="toolTip">
         <string>Description</string>
        </property>
        <property name="statusTip">
         <string>Description</string>
        </property>
        <property name="text">
         <string>Description</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="description_text">
        <property name="toolTip">
         <string>Description</string>
        </property>
        <property name="statusTip">
         <string>Description</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Entry description.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="inputMethodHints">
         <set>Qt::ImhLatinOnly|Qt::ImhNoPredictiveText</set>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="path_label">
        <property name="toolTip">
         <string>Path</string>
        </property>
        <property name="statusTip">
         <string>Path</string>
        </property>
        <property name="text">
         <string>Path</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="DevicePathView" name="device_path">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>Monospace</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Device path</string>
        </property>
        <property name="statusTip">
         <string>Device path</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Device path.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>QAbstractScrollArea::AdjustToContents</enum>
        </property>
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="horizontalScrollMode">
         <enum>QAbstractItemView::ScrollPerPixel</enum>
        </property>
        <property name="movement">
         <enum>QListView::Snap</enum>
        </property>
        <property name="resizeMode">
         <enum>QListView::Adjust</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QWidget" name="device_path_actions" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QHBoxLayout" name="paths_actions_layout" stretch="0,0,0,0,0,1">
         <property name="spacing">
          <number>1</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>1</number>
         </property>
         <property name="bottomMargin">
          <number>1</number>
         </property>
         <item>
          <widget class="QPushButton" name="path_move_up">
           <property name="maximumSize">
            <size>
             <width>40</width>
             <height>40</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Move file path up</string>
           </property>
           <property name="statusTip">
            <string>Move file path up</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Move file path up.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset theme="up">
             <normaloff>.</normaloff>.</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="path_move_down">
           <property name="maximumSize">
            <size>
             <width>40</width>
             <height>40</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Move file path down</string>
           </property>
           <property name="statusTip">
            <string>Move file path down</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Move file path down.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset theme="down">
             <normaloff>.</normaloff>.</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="path_remove">
           <property name="maximumSize">
            <size>
             <width>40</width>
             <height>40</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Remove file path</string>
           </property>
           <property name="statusTip">
            <string>Remove file path</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Remove file path.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset theme="remove">
             <normaloff>.</normaloff>.</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="path_edit">
           <property name="maximumSize">
            <size>
             <width>40</width>
             <height>40</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Edit file path</string>
           </property>
           <property name="statusTip">
            <string>Edit file path</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Edit file path.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset theme="edit">
             <normaloff>.</normaloff>.</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="path_add">
           <property name="maximumSize">
            <size>
             <width>40</width>
             <height>40</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Add file path</string>
           </property>
           <property name="statusTip">
            <string>Add file path</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Add file path.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset theme="add">
             <normaloff>.</normaloff>.</iconset>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="paths_actions_spacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="optional_data_label">
        <property name="toolTip">
         <string>Optional data</string>
        </property>
        <property name="statusTip">
         <string>Optional data</string>
        </property>
        <property name="text">
         <string>Optional</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="optional_data_format_combo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Optional data format</string>
        </property>
        <property name="statusTip">
         <string>Optional data format</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Optional data format.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <item>
         <property name="text">
          <string>BASE64</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UTF-16</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UTF-8</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>HEX</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QPlainTextEdit" name="optional_data_text">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <family>Monospace</family>
         </font>
        </property>
        <property name="toolTip">
         <string>Optional data</string>
        </property>
        <property name="statusTip">
         <string>Optional data</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Entry optional data.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>QAbstractScrollArea::AdjustToContents</enum>
        </property>
        <property name="undoRedoEnabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="attributes_label">
        <property name="toolTip">
         <string>Attributes</string>
        </property>
        <property name="statusTip">
         <string>Attributes</string>
        </property>
        <property name="text">
         <string>Attributes</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QWidget" name="attributes_box" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QGridLayout" name="attributes_box_layout" columnstretch="0,0,0,1">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <property name="spacing">
          <number>9</number>
         </property>
         <item row="0" column="0">
          <widget class="QCheckBox" name="attribute_active">
           <property name="toolTip">
            <string>Active</string>
           </property>
           <property name="statusTip">
            <string>Active</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Is entry considered for automatic boot?&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Active</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <spacer name="attribute_box_spacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="attribute_hidden">
           <property name="toolTip">
            <string>Hidden</string>
           </property>
           <property name="statusTip">
            <string>Hidden</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Hidden.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Hidden</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QCheckBox" name="attribute_force_reconnect">
           <property name="toolTip">
            <string>Force reconnect</string>
           </property>
           <property name="statusTip">
            <string>Force reconnect</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Force reconnect.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Force reconnect</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="category_label">
        <property name="toolTip">
         <string>Category</string>
        </property>
        <property name="statusTip">
         <string>Category</string>
        </property>
        <property name="text">
         <string>Category</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QComboBox" name="category_combo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Category</string>
        </property>
        <property name="statusTip">
         <string>Category</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Entry category.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <item>
         <property name="text">
          <string>Boot</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>App</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="vertical_spacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DevicePathView</class>
   <extends>QListView</extends>
   <header>devicepathview.h</header>
   <slots>
    <slot>removeCurrentRow()</slot>
    <slot>insertRow()</slot>
    <slot>moveCurrentRowUp()</slot>
    <slot>moveCurrentRowDown()</slot>
    <slot>editCurrentRow()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>description_text</sender>
   <signal>textEdited(QString)</signal>
   <receiver>BootEntryForm</receiver>
   <slot>setDescription(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>optional_data_text</sender>
   <signal>textChanged()</signal>
   <receiver>BootEntryForm</receiver>
   <slot>optionalDataEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>305</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>path_add</sender>
   <signal>clicked()</signal>
   <receiver>device_path</receiver>
   <slot>insertRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>397</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>path_remove</sender>
   <signal>clicked()</signal>
   <receiver>device_path</receiver>
   <slot>removeCurrentRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>397</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>path_move_up</sender>
   <signal>clicked()</signal>
   <receiver>device_path</receiver>
   <slot>moveCurrentRowUp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>397</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>path_move_down</sender>
   <signal>clicked()</signal>
   <receiver>device_path</receiver>
   <slot>moveCurrentRowDown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>397</x>
     <y>173</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>attribute_active</sender>
   <signal>stateChanged(int)</signal>
   <receiver>BootEntryForm</receiver>
   <slot>setAttribute(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>attribute_hidden</sender>
   <signal>stateChanged(int)</signal>
   <receiver>BootEntryForm</receiver>
   <slot>setAttribute(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>233</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>attribute_force_reconnect</sender>
   <signal>stateChanged(int)</signal>
   <receiver>BootEntryForm</receiver>
   <slot>setAttribute(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>314</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>category_combo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>BootEntryForm</receiver>
   <slot>setAttribute(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>243</x>
     <y>345</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>path_edit</sender>
   <signal>clicked()</signal>
   <receiver>device_path</receiver>
   <slot>editCurrentRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>172</x>
     <y>123</y>
    </hint>
    <hint type="destinationlabel">
     <x>243</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>optional_data_format_combo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>BootEntryForm</receiver>
   <slot>setOptionalDataFormat(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>247</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>device_path</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>device_path</receiver>
   <slot>editCurrentRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>249</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>index_text</sender>
   <signal>textEdited(QString)</signal>
   <receiver>BootEntryForm</receiver>
   <slot>setIndex(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>247</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setOptionalDataFormat(int)</slot>
  <slot>optionalDataEdited()</slot>
  <slot>setDescription(QString)</slot>
  <slot>setAttribute(int)</slot>
  <slot>setIndex(QString)</slot>
 </slots>
</ui>
